test/IdentityServer4.RavenDB.Storage.Tests/HelpersTests/DocumentStoreHelperTests.cs

```diff
@@ -1,32 +1,32 @@
-﻿using System;
-using IdentityServer4.RavenDB.Storage.Helpers;
-using Xunit;
-
-namespace IdentityServer4.RavenDB.IntegrationTests.HelpersTests
-{
-    public class DocumentStoreHelperTests
-    {
-        [Fact]
-        public void InitializeDocumentStore_Throws_WhenDatabaseNameIsNotProvided()
-        {
-            Assert.Throws<InvalidOperationException>(() => DocumentStoreHelper.InitializeDocumentStore(store =>
-            {
-                store.Database = null;
-                store.Urls = new[]
-                {
-                    "http://some-url"
-                };
-            }));
-        }
-
-        [Fact]
-        public void InitializeDocumentStore_Throws_WhenUrlsIsEmptyCollection()
-        {
-            Assert.Throws<InvalidOperationException>(() => DocumentStoreHelper.InitializeDocumentStore(store =>
-            {
-                store.Database = "some-database-name";
-                store.Urls = new string [] { };
-            }));
-        }
-    }
+﻿using System;
+using IdentityServer4.RavenDB.Storage.Helpers;
+using Xunit;
+
+namespace IdentityServer4.RavenDB.Storage.Tests.HelpersTests
+{
+    public class DocumentStoreHelperTests
+    {
+        [Fact]
+        public void InitializeDocumentStore_Throws_WhenDatabaseNameIsNotProvided()
+        {
+            Assert.Throws<InvalidOperationException>(() => DocumentStoreHelper.InitializeDocumentStore(store =>
+            {
+                store.Database = null;
+                store.Urls = new[]
+                {
+                    "http://some-url"
+                };
+            }));
+        }
+
+        [Fact]
+        public void InitializeDocumentStore_Throws_WhenUrlsIsEmptyCollection()

```

Comment:
Please add a test for null `store.Urls` as well.

Reply:
I'm afraid we can't create a reliable test for that scenario. 
`InitializeDocumentStore` accepts a configuration function that runs immediately. 
The idea behind the config function is quite straightforward - it allows `documentStore` setters to be invoked.  
In the RavenDB client code the `Urls` setter throws whenever there is an attempt to assign `null` to it:

`public string[] Urls
    {
      get => this._urls;
      set
      {
        this.AssertNotInitialized(nameof (Urls));
        if (value == null)
          throw new ArgumentNullException(nameof (value));`
          
That is why the check for `null` Urls in the `InitializeDocumentStore` function appears to be pointless as that line of code will never be reached in case the field is null. The scenario is handled by the `RavenDB` client.

Reply:
👍